include/oneapi/tbb/detail/_flow_graph_node_impl.h

```diff
@@ -746,12 +754,13 @@ class continue_input : public continue_receiver {
 #if _MSC_VER && !__INTEL_COMPILER
 #pragma warning (pop)
 #endif
-            return apply_body_bypass( continue_msg() __TBB_FLOW_GRAPH_METAINFO_ARG(message_metainfo{}) );
+            return apply_body_bypass( continue_msg() __TBB_FLOW_GRAPH_METAINFO_ARG(metainfo) );
         }
         else {
             d1::small_object_allocator allocator{};
-            typedef apply_body_task_bypass<class_type, continue_msg> task_type;
-            graph_task* t = allocator.new_object<task_type>( graph_reference(), allocator, *this, continue_msg(), my_priority );
+            typedef apply_body_task_bypass<class_type, continue_msg __TBB_FLOW_GRAPH_METAINFO_ARG(graph_task_with_message_waiters)> task_type;
+            graph_task* t = allocator.new_object<task_type>( graph_reference(), allocator, *this, continue_msg(),
+                                                             my_priority __TBB_FLOW_GRAPH_METAINFO_ARG(metainfo));
```

Comment:
I think this part should be similar with `function_node` - if the meta is empty - create an spawn `apply_body_task_bypass` with `graph_task` base class. Otherwise - create and spawn with `graph_task_with_message_waiters` task.

Reply:
Fixed